src/server/server_family.cc

```diff
@@ -2848,11 +2849,8 @@ void ServerFamily::ReplicaOfInternal(CmdArgList args, Transaction* tx, SinkReply
     builder->SendError(ec ? ec.Format() : "replication cancelled");
     return;
   }
-  // Successfully connected now we flush
-  // If we are called by "Replicate", tx will be null but we do not need
-  // to flush anything.
+
   if (on_err == ActionOnConnectionFail::kReturnOnError) {
-    Drakarys(tx, DbSlice::kDbAll);
```

Comment:
Oh you removed it. So no need to follow up on this

Reply:
:pray:

Reply:
You can double check me verifying that Flush is indeed called for all situations.

Reply:
It should be fine. We flush both in `InitiatePSync` and in `InitiateDflySync`. 

There is something that caught my eye though for `PSync (redis/valkey)`. We did call `Drakarys` before we Started the `MainReplicationFiber`. If you look within `PSync`, we call flush only if there is `diskless or disk based` replication! So... Unless I am missing something we actually `flushed` unconditionally even on `partial sync` which should be a bug ? 

Furthermore, lookg at `DflySync` the same as above applies. I would like to check if we have tests for partial sync that covers this so I can check my assumption.

Other than that, we are good here.

Reply:
That could explain failures like https://github.com/dragonflydb/dragonfly/issues/4180 which the data did not compare equally. If we flushed while doing partial replication because of a disconnect then we have data loss

Reply:
I do not think we support partial sync today